dnf/conf/config.py

```diff
@@ -243,6 +245,32 @@ def get_reposdir(self):
             dnf.util.ensure_dir(myrepodir)
         return myrepodir
 
+    def _check_remote_file(self, optname):
+        """
+        In case the option value is a remote URL, download it to the temporary location
+        and use this temporary file instead.
+        """
+        prio = self._get_priority(optname)
+        val = self._get_value(optname)
+        if isinstance(val, basestring):
+            location = urlparse.urlparse(val)
+            if location[0] in ('file', ''):
+                # just strip the file:// prefix
+                self._set_value(optname, location.path, prio)
+            else:
+                downloader = libdnf.repo.Downloader()
+                temp_fd, temp_path = tempfile.mkstemp()
+                try:
+                    downloader.downloadURL(None, val, temp_fd)
+                except RuntimeError as e:
+                    raise dnf.exceptions.ConfigError(
+                        _('URL "{}" could not be downloaded: {}').format(
```

Comment:
I'd be a bit more specific with the message (it will help if someone e.g. sees the error in the log and the URL doesn't give a hint):
```
_('Configuration file URL "{}" could not be downloaded: {}')
```

Or even refer to the `--config` CLI argument in the message?

Reply:
Not sure about --config reference, but I'll add more specific message.